core/src/main/scala/akka/persistence/jdbc/state/DurableStateQueries.scala

```diff
@@ -0,0 +1,28 @@
+package akka.persistence.jdbc.state
+
+import slick.jdbc.JdbcProfile
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+
+class DurableStateQueries(val profile: JdbcProfile, override val durableStateTableCfg: DurableStateTableConfiguration)
+    extends DurableStateTables {
+  import profile.api._
+
+  def _selectByPersistenceId(persistenceId: Rep[String]) =
+    durableStateTable.filter(_.persistenceId === persistenceId)
+
+  def _insertDurableState(row: DurableStateTables.DurableStateRow) =
+    durableStateTable += row
+
+  def _updateDurableState(row: DurableStateTables.DurableStateRow) = {
+    durableStateTable
+      .filter(r =>
+        r.persistenceId === row.persistenceId &&
+        r.seqNumber === row.seqNumber - 1)
+      .map(r => (r.statePayload, r.stateSerId, r.stateSerManifest))
+      .update((row.statePayload, row.stateSerId, row.stateSerManifest))
```

Comment:
The `upsertObject` method accepts a `tag`, but we are not using it in the update. 

We don't recommend updating tags for event sourced entities, because that breaks the projections. The right thing to do for ES is to re-tag the whole journal and restart the consumers.

However, for durable state we may uncover an interesting use case. 

We can start tagging the `User` entity with `user0`, `user1` and `user2` (using persistenceId.hashCode % 3). And we have three projections consuming those tags. 

We can then change the code to use `% 10` and we start tagging from `user0` to `user9`. 

In the beginning, we will have 10 projections running but only 3 doing real work. As we start to update the `Users`, the start to migrate to another tag slice. After some time, all 10 projections will be doing work.

Reply:
added tag in the list of columns to be updated ..

Reply:
also I guess the `stateTimestamp` column needs to be updated on upsert - @patriknw ?

Reply:
yes, the timestamp should be updated

Reply:
yes, doing that ..